data-pipeline/src/trace_exporter.rs

```diff
@@ -298,6 +345,21 @@ impl TraceExporter {
             })
     }
 
+    /// Emit a health metric to dogstatsd
+    fn emit_metric(&self, metric: HealthMetric, custom_tags: Option<Vec<&Tag>>) {
+        if let Some(flusher) = &self.dogstatsd {
+            let tags = match custom_tags {
+                None => Either::Left(&self.common_stats_tags),
+                Some(custom) => Either::Right(self.common_stats_tags.iter().chain(custom)),
+            };
```

Comment:
Yeah I definitely don't _love_ having to use it here but I couldn't find a way around it. I'm also not a huge fan of a more descriptive type since that type would only be used in exactly this one place, and would only exist to get around the type system so I can have `tags` be something that implements "Into Iterator" and avoid needing dynamic dispatch. (more context here: https://github.com/DataDog/libdatadog/pull/638#discussion_r1773354431)